src/databricks/labs/ucx/hive_metastore/mapping.py

```diff
@@ -103,14 +150,61 @@ def skip_table(self, backend: StatementExecutionBackend, schema: str, table: str
         except BadRequest as br:
             logger.error(br)
 
-    def skip_schema(self, backend: StatementExecutionBackend, schema: str):
+    def skip_schema(self, schema: str):
         # Marks a schema to be skipped in the migration process by applying a table property
         try:
-            backend.execute(f"ALTER SCHEMA `{schema}` SET DBPROPERTIES('{self.UCX_SKIP_PROPERTY}' = true)")
+            self._backend.execute(f"ALTER SCHEMA `{schema}` SET DBPROPERTIES('{self.UCX_SKIP_PROPERTY}' = true)")
         except NotFound as nf:
             if "[SCHEMA_NOT_FOUND]" in str(nf):
                 logger.error(f"Failed to apply skip marker for Schema {schema}. Schema not found.")
             else:
                 logger.error(nf)
         except BadRequest as br:
             logger.error(br)
+
+    def _get_valid_source_table(self, table: Table) -> Table | None:
+        result = self._backend.fetch(f"SHOW TBLPROPERTIES `{table.database}`.`{table.name}`")
+        for value in result:
+            if value["key"] == "upgraded_to":
```

Comment:
Bug: if the referred table in upgraded_to doesn't exist or is incorrectly migrated, it's getting skipped without a notice to a user. Raise an exception if such case occurs. 

Add integration test to cover this case's happy and unhappy path.

Call ws.tables.get() and compare the "upgraded_from" property.

Also, check skip before checking the target in upgraded_to, so that users can skip incorrectly migrated tables and unblock themselves

Reply:
Done